### ISSUE TITLE
React Context deprecated warning

### ISSUE THREAD

**Issue by mydesweb** [facebook/react#16250]
When I use this.context in a class I see a warning "Deprecated symbol used, consult docs for better alternative" in PhpStorm IDE.. it is really deprecated?

"react": "16.8.3"

**Comment by bedakb** [comment]
I'm experiencing same issue in Webstorm.
It's not deprecated (according to latest React docs) but I think It's referring to the legacy Context API, so that's why.

**Comment by gaearon** [comment]
Please report this to PhpStorm/WebStorm. This is a mistake because `this.context` *can* be a modern API. Only `childContextTypes`, `getChildContext()` and `contextTypes` (NOT `contextType`) are getting deprecated.

cc @prigara

**Comment by prigara** [comment]
Hi,
It seems to be marked as deprecated in the type definition file for React we are using. Will check if the problem is on the IDE side or on the .d.ts file. Thanks for the feedback.

**Comment by PathToLife** [comment]
![image](https://user-images.githubusercontent.com/12622625/64690270-15823e80-d4e4-11e9-8e34-edbc23776941.png)
``` /**
         * If using the new style context, re-declare this in your class to be the
         * `React.ContextType` of your `static contextType`.
         *
         * ```ts
         * static contextType = MyContext
         * context!: React.ContextType<typeof MyContext>
         * ```
         *
         * @deprecated if used without a type annotation, or without static contextType
         * @see https://reactjs.org/docs/legacy-context.html
         */```
![image](https://user-images.githubusercontent.com/12622625/64690275-19ae5c00-d4e4-11e9-969e-5129df1b0e51.png)
TLDR according to the helpful notes just make sure to declare as:  
`context!: React.ContextType<typeof AppContext>;`

**Comment by VitalyKrenel** [comment]
Hey @PathToLife! 
Do you know how to fix deprecated highlighting if the file does use plain JS only?

**Comment by mydesweb** [comment]
> Hi,
> It seems to be marked as deprecated in the type definition file for React we are using. Will check if the problem is on the IDE side or on the .d.ts file. Thanks for the feedback.

any update?

**Comment by prigara** [comment]
Hi all, I've sent a PR to the DefinitelyTyped repo to remove the deprecation warning: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/39177

**Comment by mendoncanorma** [comment]
@prigara , The deprecated warning is still seen and we are on webstorm 2020.1
